### ISSUE TITLE
Data race

### ISSUE THREAD

**Issue by mightyguava** [cirruslabs/echelon#14]
I have some unit tests on some code that uses echelon and it spit this out with `-race`.

Looks like the `len(node.Children)` at https://github.com/cirruslabs/echelon/blob/master/renderers/internal/node/echelone_node.go#L134 needs to be protected by mutex.

```
==================
WARNING: DATA RACE
Write at 0x00c0001386a8 by goroutine 10:
  github.com/cirruslabs/echelon/renderers/internal/node.(*EchelonNode).FindOrCreateChild()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/internal/node/echelone_node.go:194 +0x2bd
  github.com/cirruslabs/echelon/renderers.findScopedNode()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/interactive.go:43 +0xb0
  github.com/cirruslabs/echelon/renderers.(*InteractiveRenderer).RenderScopeStarted()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/interactive.go:49 +0x67
  github.com/cirruslabs/echelon.(*Logger).streamEntries()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/logger.go:46 +0x182

Previous read at 0x00c0001386a8 by goroutine 9:
  github.com/cirruslabs/echelon/renderers/internal/node.(*EchelonNode).fancyTitle()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/internal/node/echelone_node.go:134 +0x82
  github.com/cirruslabs/echelon/renderers/internal/node.(*EchelonNode).Render()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/internal/node/echelone_node.go:100 +0x6a
  github.com/cirruslabs/echelon/renderers.(*InteractiveRenderer).DrawFrame()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/interactive.go:94 +0x1aa
  github.com/cirruslabs/echelon/renderers.(*InteractiveRenderer).StartDrawing()
      /data/app/kochiku-worker/go/pkg/mod/github.com/cirruslabs/echelon@v1.3.2/renderers/interactive.go:75 +0x74
=================
```

**Comment by fkorotkov** [comment]
Fixing in #15

**Comment by fkorotkov** [comment]
Should be fixed in 1.4.1

**Comment by fkorotkov** [comment]
Thank you for reporting! 🙌
